manifests/resource/server.pp

```diff
@@ -171,32 +171,32 @@
   Optional[Variant[String, Boolean]] $ssl_cert                                   = undef,
   Optional[String] $ssl_client_cert                                              = undef,
   String $ssl_verify_client                                                      = 'on',
-  Optional[String] $ssl_dhparam                                                  = $nginx::ssl_dhparam,
+  Optional[String] $ssl_dhparam                                                  = undef,
```

Comment:
Is this change ok, as if `$nginx::ssl_dhparam` is defined it will now get written to `nginx.conf` or is this considered a backward-incompatible change even if nginx should be working as before this change?

Reply:
I do not think this change is backwards-incompatible, because the value was ```undef```before and can be overwritten any time. However, you are using a different data type in ```init.pp``` and ```server.pp```. 

Unfortunately I can not have a look at the failing Travis CI tests, because the website is under maintenance.

Reply:
I don't mind a breaking change for this since it makes sense IMHO.

Reply:
@dhoppe i made the settings in init.pp more strict as these are new parameters. I didn't want to break anything with my changes if somebody is already using `nginx::resource::server`.

There are also `$ssl_stapling` and `$ssl_stapling_verify` which are `Enum['on', 'off']` on init.pp and `Boolean` in server.pp
Unifying them will be definitely a breaking-change and should probably be done in a separate ticket?

If there is any new data type to strict just tell me and i will change it :) I would like to see this change as soon as possible released as we would like to move some ssl-settings vom vhost/servers to the global config in our infrastructure.